### PR TITLE
feat(behavior_path_planner): cog centered path shape planning

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description
No behavior change with default parameters
Added a option where the CoG instead of back wheel center will follow the centerline, which is false by default.

**Logic**
image

**Result**
before
image

after
image

## Test performed
Nothing due to no behavior change.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
@TakaHoribe will review this PR.

File: common/motion_utils/include/motion_utils/trajectory/path_with_lane_id.hpp [hunk 1]
```diff
@@ -99,6 +99,11 @@ inline size_t findNearestSegmentIndexFromLaneId(
 
   return nearest_idx;
 }
+
+// NOTE: rear_to_cog is supposed to be positive
```
Comment (TakaHoribe) [inline common/motion_utils/include/motion_utils/trajectory/path_with_lane_id.hpp:103]: This function is hard to understand for users. Please write a detailed description.

```suggestion
// @brief Calculates the path to be followed by the rear wheel center in order to make the vehicle center follow the input path
// @param [in] path with position to be followed by the center of the vehicle
// @param [out] PathWithLaneId to be followed by the rear wheel center follow to make the vehicle center follow the input path
// NOTE: rear_to_cog is supposed to be positive
```

File: common/motion_utils/src/trajectory/path_with_lane_id.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  for (size_t i = 0; i < path.points.size(); ++i) {
+    // calculate beta, which is CoG's velocity direction
+    const double beta = std::atan(rear_to_cog * curvature_vec.at(i));
+
+    // apply beta to CoG pose
+    geometry_msgs::msg::Pose cog_pose_with_beta;
+    cog_pose_with_beta.position = tier4_autoware_utils::getPoint(path.points.at(i));
+    cog_pose_with_beta.orientation =
+      tier4_autoware_utils::createQuaternionFromYaw(yaw_vec.at(i) - beta);
```
Comment (TakaHoribe) [inline common/motion_utils/src/trajectory/path_with_lane_id.cpp:39]: Is this really ok with minus? 

```
yaw_vec.at(i) - beta
```

Comment (takayuki5168) [reply common/motion_utils/src/trajectory/path_with_lane_id.cpp:39]: In the case of the logic figure in the description, `beta` is a negative value, and we wanna rotate the yaw angle anti-clockwisely.
Therefore, minus is correct I think.

**Comment by takayuki5168** [comment]
pre-commit.ci run

**Comment by takayuki5168** [comment]
pre-commit.ci run

**Review by TakaHoribe — APPROVED** [review]
LGTM

**Pull request merged**